benchmarks/graphics_pipeline/GraphicsBenchmarkApp.cpp

```diff
@@ -872,18 +875,21 @@ void GraphicsBenchmarkApp::ProcessKnobs()
 {
     // Detect if any knob value has been changed
     // Note: DigestUpdate should be called only once per frame (DigestUpdate unflags the internal knob variable)!
-    const bool allTexturesTo1x1KnobChanged = pAllTexturesTo1x1->DigestUpdate();
-    const bool alphaBlendKnobChanged       = pAlphaBlend->DigestUpdate();
-    const bool depthTestWriteKnobChanged   = pDepthTestWrite->DigestUpdate();
-    const bool enableSpheresKnobChanged    = pEnableSpheres->DigestUpdate();
+    const bool allTexturesTo1x1KnobChanged    = pAllTexturesTo1x1->DigestUpdate();
+    const bool alphaBlendKnobChanged          = pAlphaBlend->DigestUpdate();
+    const bool depthTestWriteKnobChanged      = pDepthTestWrite->DigestUpdate();
+    const bool enableSpheresKnobChanged       = pEnableSpheres->DigestUpdate();
+    const bool sphereInstanceCountKnobChanged = pSphereInstanceCount->DigestUpdate();
 
     const bool enableSpheres = pEnableSpheres->GetValue();
     // If the LOD is empty, assuming we skipped the setup for all sphere resources at start
     // So need to do the intial setup for all resources here
-    const bool spheresAreSetUp               = mSpheresAreSetUp;
-    const bool updateSphereDescriptors       = spheresAreSetUp && allTexturesTo1x1KnobChanged;
-    const bool setupSphereResourcesAndMeshes = (!spheresAreSetUp) && enableSpheres;
-    const bool rebuildSpherePipeline         = setupSphereResourcesAndMeshes || (spheresAreSetUp && (alphaBlendKnobChanged || depthTestWriteKnobChanged));
+    const bool spheresAreSetUp         = mSpheresAreSetUp;
+    const bool requireMoreSpheres      = sphereInstanceCountKnobChanged && (pSphereInstanceCount->GetValue() > mInitializedSpheres);
+    const bool updateSphereDescriptors = spheresAreSetUp && allTexturesTo1x1KnobChanged;
+    const bool setupSphereResources    = (!spheresAreSetUp) && enableSpheres;
+    const bool setupSphereMeshes       = ((!spheresAreSetUp) || requireMoreSpheres) && enableSpheres;
+    const bool rebuildSpherePipeline   = setupSphereMeshes || (spheresAreSetUp && (alphaBlendKnobChanged || depthTestWriteKnobChanged));
```

Comment:
Do we need to rebuild pipelines if there was an increase in the number of spheres but nothing else changed?

Reply:
Not directly related to this PR, but I wonder if it should also take into account the other knob changes (vertex buffer format, layout, and shaders) or if it can be ommitted altogether since `GetSpherePipeline()` in `RecordCommandBufferSpheres` will create the necessary pipeline anyway

Reply:
That is a good point, actually we don't need to rebuild pipelines. so I applied your proposed clean up from the other comment.